### PR TITLE
New plugin for dumping .torrent info

### PR DESCRIPTION
**Pull request by TheGoblinHero**
Regularly I need to export info about torrent (all filenames, exact size in bytes, etc) to the text format. So I wrote a simple plugin:
dump
The plugin uses external tool to dump the torrent. Personally I use [old dumptorrent utility with several patches](https://github.com/TheGoblinHero/dumptorrent), but it's configurable, so you can use any utility you want.

### CODE REVIEW

File: plugins/dump/plugin.info [hunk 1]
```diff
@@ -0,0 +1,4 @@
+plugin.description: This plugin dumps .torrent file information to the console
+plugin.author: TheGoblinHero
+plugin.version: 1.0
+plugin.dependencies: _task
```
Comment (stickz) [inline plugins/dump/plugin.info:4]: Please add an external error for `dumptorrent`.
```
rtorrent.external.error: dumptorrent
```

File: plugins/dump/lang/ru.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * PLUGIN DUMP
+ *
+ * Russian language file.
+ *
+ * Author:
+ */
+
+theUILang.dumpTorrent		= "Получить дамп...";
+
+thePlugins.get("dump").langLoaded();
```
Comment (stickz) [inline plugins/dump/lang/ru.js:11]: Please add new line at end of file.

File: plugins/dump/lang/en.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+ * PLUGIN DUMP
+ *
+ * English language file.
+ *
+ * Author:
+ */
+
+theUILang.dumpTorrent		= "Dump .torrent...";
+
+thePlugins.get("dump").langLoaded();
```
Comment (stickz) [inline plugins/dump/lang/en.js:11]: Please add new line at end of file.

File: plugins/dump/conf.php [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<?php
+
+$pathToExternals['dumptorrent'] = '';	// Path to the program that dumps .torrent file information
+$arguments = "-v";			// Program arguments
```
Comment (stickz) [inline plugins/dump/conf.php:4]: Please add new line at end of file.

File: plugins/dump/conf.php [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?php
+
```
Comment (stickz) [inline plugins/dump/conf.php:2]: Please add a comment here, where we can find the external dependency required.

**Review by stickz — CHANGES_REQUESTED** [review]
Please submit to `develop` branch instead of `master`. We will release with `ruTorrent v4.4`. The wiki also as public permissions to edit. You can add this plugin in there.

**Comment by TheGoblinHero** [comment]
Fixed. [Wiki page is added](https://github.com/Novik/ruTorrent/wiki/PluginDump). Should I create another PR to submit to develop?

**Comment by stickz** [comment]
LGTM! Thanks!

**Comment by buggsi** [comment]
Useful plugin, I started testing v5.1beta and the plugin is included.

**Comment by jegue** [comment]
if  dumptorrent not installed 
follow this

sudo apt-get install gcc make
wget "from github source"
tar -xvf dumptorrent-1.x.tar.gz
cd dumptorrent-1.x
make
sudo cp dumptorrent /usr/local/bin

**Pull request merged**